crates/swc_ecma_compat_es2015/src/classes/mod.rs

```diff
@@ -31,14 +30,25 @@ use self::{
 mod constructor;
 mod prop_name;
 
-pub fn classes<C>(comments: Option<C>, config: Config) -> impl Fold + VisitMut
+pub fn classes<C>(
+    TransformContext {
+        assumptions,
+        comments,
+        ..
+    }: TransformContext<C>,
+) -> impl Fold + VisitMut
 where
-    C: Comments,
+    C: Comments + Clone,
 {
     as_folder(Classes {
         in_strict: false,
         comments,
-        config,
+        config: Config {
```

Comment:
The constructor drops `Assumptions` early to avoid making difference of memory access